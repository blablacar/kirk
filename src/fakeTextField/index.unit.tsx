import React from 'react'
import renderer from 'react-test-renderer'

import FakeTextField from 'fakeTextField'

describe('FakeTextField', () => {
  it('Should have the proper default props', () => {
    const wrapper = renderer.create(<FakeTextField>Hello world</FakeTextField>)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should have the text ellipsed', () => {
    const wrapper = renderer.create((
      <FakeTextField ellipsis>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        Donec nec tristique sapien, eu placerat justo. Donec tempor,
        risus ac cursus fringilla, lorem ipsum facilisis tortor, vel
        molestie sapien justo nec orci.
      </FakeTextField>
    ))
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should support add-ons', () => {
    const wrapper = renderer.create((
      <FakeTextField addOn={<div>Add-on</div>}>
        Hello world
      </FakeTextField>
    ))
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should support simple links', () => {
    const wrapper = renderer.create((
      <FakeTextField href="#foo">
        Click me
      </FakeTextField>
    ))
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('Should support component links', () => {
    const wrapper = renderer.create((
      <FakeTextField href={<a href="#bar" />}>
        Click me
      </FakeTextField>
    ))
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
