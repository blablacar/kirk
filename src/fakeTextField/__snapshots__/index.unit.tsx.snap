// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FakeTextField Should have the proper default props 1`] = `
<div
  className="fakeTextField"
>
  <div
    className="jsx-3143559470 fakeTextField-label"
  >
    Hello world
  </div>
</div>
`;

exports[`FakeTextField Should have the text ellipsed 1`] = `
<div
  className="fakeTextField"
>
  <div
    className="jsx-3143559470 fakeTextField-label fakeTextField-label--ellipsis"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec nec tristique sapien, eu placerat justo. Donec tempor, risus ac cursus fringilla, lorem ipsum facilisis tortor, vel molestie sapien justo nec orci.
  </div>
</div>
`;

exports[`FakeTextField Should support add-ons 1`] = `
<div
  className="fakeTextField"
>
  <div>
    Add-on
  </div>
  <div
    className="jsx-3143559470 fakeTextField-label"
  >
    Hello world
  </div>
</div>
`;

exports[`FakeTextField Should support component links 1`] = `
<a
  className="fakeTextField"
  href="#bar"
>
  <div
    className="jsx-3143559470 fakeTextField-label"
  >
    Click me
  </div>
</a>
`;

exports[`FakeTextField Should support simple links 1`] = `
<a
  className="fakeTextField"
  href="#foo"
>
  <div
    className="jsx-3143559470 fakeTextField-label"
  >
    Click me
  </div>
</a>
`;
